### DISCUSSION TITLE
Prefilling SWR cache with the App directory

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/swr#2692] [Q&A] [answered]
The docs have a [very good example on how to use SWR with SSG and SSR](https://swr.vercel.app/docs/with-nextjs#pre-rendering-with-default-data), however there's no mention of how to use `swr` with the new app directory.

Vercel now suggests passing data to the page [like this](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#accessing-request-object), but I'm honestly stumped on how it would work with `SWRConfig`. 

Are there any docs/help for this?

**Comment by promer94** [comment, marked as answer]
https://swr.vercel.app/docs/with-nextjs#app-dir
